### PR TITLE
Fix failing e2e test for cow amms

### PR DESCRIPTION
**Pull request by sunce86**
@MartinquaXD I believe you wrote the test, do you remember why this specific pool was added to the list?

### CODE REVIEW

**Comment by MartinquaXD** [comment]
Given that this error seemingly started with https://github.com/cowprotocol/services/pull/2871 I would say this indicates that we had a pool which traded cow and some other token but we couldn't compute a price for the other token so now we also drop the cow price although we have it.

**Review by squadgazzz — APPROVED** [review]
Nice, now we can say the change is covered by tests 😄

File: crates/autopilot/src/solvable_orders.rs [hunk 1]
```diff
@@ -253,10 +253,17 @@ impl SolvableOrdersCache {
...
-                cow_amm
+                let prices_exist = cow_amm
                     .traded_tokens()
                     .iter()
-                    .all(|token| prices.contains_key(token))
+                    .all(|token| prices.contains_key(token));
+                if !prices_exist {
+                    tracing::debug!(
+                        cow_amm = ?cow_amm.address(),
+                        "prices missing"
```
Comment (MartinquaXD) [inline crates/autopilot/src/solvable_orders.rs:263]: log could be more explicit
Maybe: `omitted from auction due to missing prices`

**Review by MartinquaXD — COMMENTED** [review]
@sunce86 could you double check dropping which pool causes the cow price to miss as well just to be sure this is the root cause?

File: crates/autopilot/src/solvable_orders.rs (hunk 1 above)
Comment (sunce86) [reply crates/autopilot/src/solvable_orders.rs]: fixed log, added missing token

**Comment by sunce86** [comment]
> @sunce86 could you double check dropping which pool causes the cow price to miss as well just to be sure this is the root cause?

Cow token doesn't have price, I am not sure if this is expected in this test or not

2024-08-13T08:49:19.511Z  WARN request{id="41"}:/quote{solver=test_solver}: driver::infra::observe: failed to quote order order=Order { tokens: Tokens { sell: TokenAddress(ContractAddress(0xdef1ca1fb7fbcdc777520aa7f396b4e015f497ab)), buy: TokenAddress(ContractAddress(0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2)) }, amount: TargetAmount(1000000000000000000), side: Buy, deadline: Deadline { driver: 2024-08-13T08:49:24.004674131Z, solvers: 2024-08-13T08:49:23.104904229Z } } err=QuotingFailed(NoSolutions)

2024-08-13T08:49:19.694Z DEBUG solvable_orders_cache: autopilot::solvable_orders: omitted from auction due to missing prices cow_amm=0xbeef5afe88ef73337e5070ab2855d37dbf5493a4 token=0xdef1ca1fb7fbcdc777520aa7f396b4e015f497ab

**Comment by MartinquaXD** [comment]
yeah, that is expected. See [here](https://github.com/cowprotocol/services/pull/2883/files#diff-5ac885a2341d119cb2910be92647a90374783b5694ac212e4f8392384bf42b3cR562-R564).

**Review by MartinquaXD — APPROVED** [review]

**Comment by sunce86** [comment]
> yeah, that is expected. See [here](https://github.com/cowprotocol/services/pull/2883/files#diff-5ac885a2341d119cb2910be92647a90374783b5694ac212e4f8392384bf42b3cR562-R564).

Aaaah yes just found it! Thanks!

**Pull request merged**